main.py

```diff
@@ -58,11 +65,19 @@ def chart(chart_type):
     display = request.args.get("display")
     url = request.args.get("formatted_results_location", None)
     if url is None:
-        raise Exception("No formatted_results_location")
-    formatted_results = requests.get(url, headers=request.headers).text
-    # TODO check response code
+        raise ResultsLocationRequired
+    response = requests.get(url, headers=request.headers)
+    if not response.status_code == 200:
+        message = "Formatted resulst unavailable at {}. Got{} response".format(
```

Comment:
might be nice to have a space after `Got` here

Reply:
yep